platform/darwin/src/MGLOfflineStorage.h

```diff
@@ -195,6 +195,13 @@ typedef void (^MGLOfflinePackRemovalCompletionHandler)(NSError * _Nullable error
  KVO change notifications on the shared offline storage object’s `packs` key
  path. Removals from that array result in an `NSKeyValueChangeRemoval` change.
  
+ @note When you remove an offline pack, any resources that are required by that
+    pack, but not other packs, become eligible for removal from the offline
+    database. Because the offline database is also used as a general purpose
```

Comment:
You’re right that it can be awkward to talk in more general terms about what, at the end of the day, is an implementation detail. I suppose as long as we don’t say anything about SQL, SQLite, or Core Data, my concerns are a bit overblown. Merge away.